desc/objectives/_coils.py

```diff
@@ -982,15 +987,15 @@ def build(self, use_jit=True, verbose=1):
             "Plasma/Surface grid includes interior points, should be rho=1.",
         )
 
-        self._dim_f = coils.num_coils
+        self._dim_f = coil.num_coils
```

Comment:
either change name `coils` or add a check in this init to say it must have a coilset passed in